HSE-Project-RAG/bench.py

```diff
@@ -0,0 +1,222 @@
+from qdrant_client import QdrantClient, models
+from qdrant_client.models import VectorParams, Distance
+from sentence_transformers import SentenceTransformer
+
+
+class Bench:
+    """
+    A class for working with text embeddings and storing them in Qdrant.
+
+    Attributes:
+        model_name (str): Name of the model used for generating embeddings.
+        device (str): Device for computation (e.g., 'cuda' or 'cpu').
+        client (QdrantClient): Client for interacting with Qdrant.
+        dataset (pd.DataFrame): Dataset containing texts and metadata.
+        sparse_model_list (list): List of models for sparse embeddings.
+        dense_model_list (list): List of models for dense embeddings.
+        sparse (bool): Flag indicating whether sparse embeddings are used.
+    """
+
+    def __init__(self,
+                 model_name="all-MiniLM-L6-v2",
+                 device="cuda",
+                 m = 16,
+                 ef_construct = 100,
+                 full_scan_threshold = 0):
+        """
+        Initializes the Bench class.
+
+        Args:
+            model_name (str): Name of the model for generating embeddings. Defaults to "all-MiniLM-L6-v2".
+            device (str): Device for computation. Defaults to "cuda".
+        """
+        self.model_name = model_name
+        self.device = device
+        self.client = QdrantClient()
+        self.dataset = pd.read_parquet(
+            'hf://datasets/neural-bridge/rag-dataset-12000/data/train-00000-of-00001-9df3a936e1f63191.parquet'
+        ).dropna().reset_index(drop=True)
+
+        self.sparse_model_list = ['prithvida/Splade_PP_en_v1', "Qdrant/bm25"]
+        self.dense_model_list = ["all-MiniLM-L6-v2", 'intfloat/multilingual-e5-small', 'all-mpnet-base-v2']
+        self.sparse = False
+        self.m = m
+        self.full_scan_threshold = full_scan_threshold
+        self.ef_construct = ef_construct
+
+    def prepare_collection(self):
+        """
+        Prepares the Qdrant collection based on the type of model (sparse or dense).
+        """
+        if self.model_name in self.dense_model_list:
+            self.prepare_dense_collection()
+        elif self.model_name in self.sparse_model_list:
+            self.sparse = True
+            self.prepare_sparse_collection()
+
+    def prepare_dense_collection(self):
+        """
+        Prepares a collection with dense embeddings.
+        """
+        self.model = SentenceTransformer(self.model_name, device=self.device)
+
+        if '/' in self.model_name:
+            self.model_name = self.model_name.split('/')[-1]
+
+        if f"{self.model_name}_vectors.npy" not in glob("*"):
+            vectors = self.model.encode(self.dataset['context'], show_progress_bar=True)
+            np.save(f"{self.model_name}_vectors.npy", vectors, allow_pickle=False)
+        else:
+            vectors = np.load(f"{self.model_name}_vectors.npy")
+
+        if self.client.collection_exists(collection_name="test"):

```

Comment:
"test" should be defined as a constant somewhere in a config file or declared as a class variable